backends/platform/3ds/osystem-graphics.cpp

```diff
@@ -150,6 +154,9 @@ void OSystem_3DS::initSize(uint width, uint height,
 	_gameHeight = height;
 	_gameTopTexture.create(width, height, _pfGameTexture);
 	_overlay.create(getOverlayWidth(), getOverlayHeight(), _pfGameTexture);
+	_gameMagTexture.create(_magWidth, _magHeight, _pfGameTexture);
```

Comment:
It would be better if the magnify mode did not use an extra texture / copy of the game screen. The 3DS is a low power device, the more processing power remains available to the game engine, the better.

`_gameTopTexture` already contains a full resolution version of the game screen. What needs to happen is showing the magnified part of the texture. This can be done by setting the UV coordinates of the vertices when rendering the sprite:
https://github.com/scummvm/scummvm/blob/af4e15ce3f6b4a22b9f7a4246e9e4e918042a085/backends/platform/3ds/sprite.cpp#L79

Reply:
Because _gameBottomTexture pulls pixel data from _gameTopTexture, any changes to the latter's UV coords would affect that of the former as well. Because only the top screen's position should change when magnified, I opted to instead manipulate _gameTopTexture's scale and position.